lib/usermanager.js

```diff
@@ -340,36 +340,36 @@ exports = module.exports = {
   },
 
   /**
-    * handle a users unshared courses on delete
-  */
+   * Handle a users courses on delete
+   */
   handleUserCoursesOnDelete: function(userId, courseOption, done) {
     var pluginManager = pluginmanager.getManager();
     var plugins = pluginManager.getPlugins();
     var CourseContent = require(plugins.content.course.fullPath);
-
     var courseContent = new CourseContent();
-    courseContent.retrieve({ createdBy: userId, _isShared: false }, {}, function (err, results) {
-
-      switch (courseOption) {
-        case 'transfer':
-          var currentUser = this.getCurrentUser();
-          async.each(results, function(model, cb) {
-            courseContent.update({ _id: model._id }, { createdBy: currentUser._id }, cb);
-          }, done);
-          break;
-        case 'delete':
-          async.each(results, function(model, cb) {
-            courseContent.destroy({ _id: model._id }, false, cb);
-          }, done);
-          break;
-        case 'share':
-          async.each(results, function(model, cb) {
-            courseContent.update({ _id: model._id }, { _isShared: true }, cb);
-          }, done);
-          break;
-        default:
-          done(null);
-      }
+    if (courseOption === null) {
+      return done(null);
+    }
+    courseContent.retrieve({ createdBy: userId }, {}, function (err, results) {
+      var currentUser = this.getCurrentUser();
+      async.eachSeries(results, function(model, cb) {
+        // Transfer ownership of all courses to current user
+        courseContent.update({ _id: model._id }, { createdBy: currentUser._id }, function (err, course) {
+          if (!course._isShared && course._shareWithUsers.length === 0) {
+            // Delete private courses if this option was set
+            if (courseOption === 'delete') {
+              courseContent.destroy({ _id: course._id }, false, cb);
+              return;
+            }
+            // Share private courses with everyone if this option was set
+            if (courseOption === 'share') {
+              courseContent.update({ _id: course._id }, { _isShared: true, _shareWithUsers: course._shareWithUsers.pull(currentUser._id) }, cb);
+              return;
+            }
+          }
+          courseContent.update({ _id: course._id }, { _shareWithUsers: course._shareWithUsers.pull(currentUser._id) }, cb);
```

Comment:
Suggest adding a comment here, as you've done for the statements, to make it crystal clear what is happening.